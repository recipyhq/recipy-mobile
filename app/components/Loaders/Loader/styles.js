import EStyleSheet from 'react-native-extended-stylesheet';

const styles = EStyleSheet.create({
  modalBackground: {
    flex: 1,
    alignItems: 'center',
    flexDirection: 'column',
    justifyContent: 'space-around',
    backgroundColor: '#00000040',
  },
  activityIndicatorWrapper: {
    backgroundColor: '#FFFFFF',
    height: 100,
    width: 100,
    borderRadius: 10,
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-around',
  },
});

export default styles;
