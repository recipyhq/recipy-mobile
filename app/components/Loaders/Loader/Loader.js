import React from 'react';
import PropTypes from 'prop-types';
import {
  View,
  Modal,
  ActivityIndicator,
} from 'react-native';
import styles from './styles';
import colors from '../../../config/colors';

const Loader = ({ isLoading }) => (
  <Modal
    transparent
    visible={isLoading}
    onRequestClose={() => {}}
  >
    <View style={styles.modalBackground}>
      <View style={styles.activityIndicatorWrapper}>
        <ActivityIndicator
          size="small"
          color={colors.primaryOrange}
        />
      </View>
    </View>
  </Modal>
);

Loader.propTypes = {
  isLoading: PropTypes.bool.isRequired,
};

export default Loader;
