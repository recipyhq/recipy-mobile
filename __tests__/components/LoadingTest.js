import 'react-native';
import React from 'react';

import renderer from 'react-test-renderer';
import Loader from '../../app/components/Loaders/Loader/Loader';

it('renders correctly', () => {
  const tree = renderer.create(
    <Loader isLoading />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
