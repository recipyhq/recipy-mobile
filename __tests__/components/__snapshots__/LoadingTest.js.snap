// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View>
    <View>
      <ActivityIndicator
        animating={true}
        color="#FF9A3B"
        hidesWhenStopped={true}
        size="small"
      />
    </View>
  </View>
</Modal>
`;
